### ISSUE TITLE
/etc/path changes PATH variable after .zshenv

### ISSUE THREAD

**Issue by odysseus0** [robbyrussell/oh-my-zsh#6600]
According to the current initialization of oh-my-zsh, the paths in /etc/path will be appended to the beginning of the PATH variable **after** the paths in .zshenv are added.

Such behavior causes great problem when you want some other directories be searched first before the 
/usr/local/bin
/usr/bin
/bin
/usr/sbin
/sbin
directories in /etc/path.

It may sound rare, but there is a very practical case. Consider the GNU coreutils package that is often installed through Homebrew in OS X. Many developers want them to replace the default BSD utilities installed in Mac OS, not just in interactive shell, but also in shell script. 

However, if the paths in /etc/path are added after .zshenv, there is no way for the local user to let GNU coreutils takes priority.

Fundamentally, such order of execution does not make sense from a design point of view. The .zshenv file is designed to let the local user customize the PATH variable. Therefore, it should be executed after  /etc/path, not before.

In conclusion, I would strongly recommend .zshenv be executed in initialization after the paths in /etc/path being added to the PATH variable.

**Comment by philbarr** [comment]
Think this might also be the issue I'm seeing:
Open a terminal.
Type zsh
Type echo $PATH # note the output
Type zsh
Type echo $PATH # note the output is different than previous output

**Comment by WinglessPegasus** [comment]
That is because every time you execute zsh, it will append to the PATH variable.

**Comment by mahmoudimus** [comment]
@WinglessPegasus did you get around this issue? I agree w/ your use case as well.

**Comment by odysseus0** [comment]
@mahmoudimus I got around this issue by editing /etc/path, but we do not always have root permission. I would still consider this a flaw in the design. What do you think?

**Comment by mcornella** [comment]
OMZ doesn't do `$PATH` changes, this is not something we can fix.
